### PR TITLE
#task-2 - serve SPA in AWS

### PR DESCRIPTION
**Pull request by xlegoleg**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -9,7 +9,21 @@
     "lint": "ng lint",
     "e2e": "ng e2e",
     "prepare": "husky install",
-    "lint:format": "npx --no-install lint-staged && npx --no-install pretty-quick --staged"
+    "lint:format": "npx --no-install lint-staged && npx --no-install pretty-quick --staged",
+    "deploy:s3": "serverless client deploy --no-config-change --no-policy-change --no-cors-change",
+    "build:deploy": "yarn build && yarn deploy:s3",
+    "serverless-single-page-app-plugin": "file:./serverless-single-page-app-plugin",
```
Comment (Guria) [inline package.json:15]: There is no sense in this record in scripts section.

File: package.json [hunk 2]
```diff
@@ -25,6 +39,8 @@
     "@angular/router": "~12.0.0",
     "bootstrap": "^4.6.0",
     "rxjs": "~6.6.0",
+    "serverless-finch": "^4.0.0",
+    "serverless-single-page-app-plugin": "file:./serverless-single-page-app-plugin",
```
Comment (Guria) [inline package.json:43]: You can put custom plugin in `.serverless_plugins` folder. So you will have no need to install it.
https://github.com/serverless/serverless/issues/6017#issuecomment-503139339

File: serverless.yml [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+        CloudFrontOriginAccessIdentityConfig:
+          Comment: Access identity between CloudFront and S3 bucket
+
+    ## Specifying the CloudFront Distribution to server your Web Application
+    WebAppCloudFrontDistribution:
+      Type: AWS::CloudFront::Distribution
+      Properties:
+        DistributionConfig:
+          Origins:
+            - DomainName: ${self:custom.s3BucketName}.s3.amazonaws.com
```
Comment (Guria) [inline serverless.yml:163]: There is a way to get full domain name of bucket without manual constructing it:
`- DomainName: !GetAtt WebAppS3Bucket.RegionalDomainName`

File: serverless-single-page-app-plugin/package.json [hunk 4]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (Guria) [inline serverless-single-page-app-plugin/package.json:1]: There will be no need in `package.json` here if you put plugin in `.serverless_plugins` folder.

**Review by Guria — APPROVED** [review]
**My evaluation: 5** (Evaluation form is submitted)

:heavy_check_mark: suggested plugins are added and configured
:heavy_check_mark: The app can be built and deployed by running npm script command
:heavy_check_mark: Link to CloudFront website is provided in README
:heavy_check_mark: Link to S3 bucket without public access is provided in README
:heavy_check_mark: PR with all changes is submitted in the YOUR OWN repository
:heavy_check_mark: PR submitted for review

---
Extra notes:
I have found that suggested plugins in our course are not optimal.
You can find my post in Teams channel where I suggest to use:
* https://www.serverless.com/plugins/serverless-s3-sync
* https://github.com/aghadiry/serverless-cloudfront-invalidate
* custom plugin to output cloudfront domain name

You may also check my review notes which are not affecting my final evaluation.
Feel free to reach me if you face with any problems with my suggestions.

**Comment by cemgokce** [comment]
My evaluation criteria point is 5 (form is submitted)

Everything looks good.

**Comment by vkhlystun** [comment]
My evaluation: 5 (Evaluation form is submitted)

**Pull request merged**